internal/clients/provider_config.go

```diff
@@ -38,11 +39,17 @@ const (
 	// authentication types
 	authKeyIRSA        = "IRSA"
 	authKeyWebIdentity = "WebIdentity"
+	authKeyUpbound     = "Upbound"
 	// authKeySAML        = "SAML"
 
 	envWebIdentityTokenFile = "AWS_WEB_IDENTITY_TOKEN_FILE"
 	errRoleChainConfig      = "failed to load assumed role AWS config"
 	errAWSConfig            = "failed to get AWS config"
+	errAWSConfigIRSA        = "failed to get AWS config using IAM Roles for Service Accounts"
+	errAWSConfigWebIdentity = "failed to get AWS config using web identity token"
+	errAWSConfigUpbound     = "failed to get AWS config using Upbound identity"
+
+	upboundProviderIdentityTokenFile = "/var/run/secrets/upbound.io/provider/token"
```

Comment:
This is currently not something that can be modified and it will always have the same path (similar to Kubernetes service account tokens). I think it is possible that we provide some level of customization here in the future, but given that the user can’t control the volume mount today 👍🏻